cpp/src/arrow/compute/kernels/aggregate.cc

```diff
@@ -50,14 +50,29 @@ class ManagedAggregateState {
 };
 
 Status AggregateUnaryKernel::Call(FunctionContext* ctx, const Datum& input, Datum* out) {
-  if (!input.is_array()) return Status::Invalid("AggregateKernel expects Array datum");
+  if (input.is_arraylike()) {
+    auto state = ManagedAggregateState::Make(aggregate_function_, ctx->memory_pool());
+    if (!state) return Status::OutOfMemory("AggregateState allocation failed");
 
-  auto state = ManagedAggregateState::Make(aggregate_function_, ctx->memory_pool());
-  if (!state) return Status::OutOfMemory("AggregateState allocation failed");
-
-  auto array = input.make_array();
-  RETURN_NOT_OK(aggregate_function_->Consume(*array, state->mutable_data()));
-  RETURN_NOT_OK(aggregate_function_->Finalize(state->mutable_data(), out));
+    if (input.is_array()) {
```

Comment:
I propose the following small-ish refactor to make it more readable. Just wrap the single array in a single element vector and maintain the loop?

Reply:
Is there an elegant way to put the chunks vector into a local variable without copying the vector and also reference the single-element vector? One is a reference, the other would be a local instance.